### ISSUE TITLE
Client:connect() does not return anything

### ISSUE THREAD

**Issue by thebenign** [camchenry/sock.lua#3]
The documentation states that Client:connect() will return a boolean value indicating whether the connection was successful or not. It currently returns nil in every case.

**Comment by camchenry** [comment]
You're right, and the documentation has been corrected in some of the more recent updates. (Which will come soon!) The actual connection event doesn't occur until the next time the client and server are updated. So, there's no way to tell if a connection succeeded at the time the connect method is called.

**Comment by thebenign** [comment]
Thanks for the reply! I did notice that when I looked through the code. It makes sense with how enet handles peer creation / connection. Do you plan to implement this feature some other way?

**Comment by camchenry** [comment]
For now, you can just set some connected state when the "connect" event goes off.

``` lua
-- client code
connected = false

client:connect()

client:on("connect", function()
    connected = true
end)
```
